File: nas_spec/code_samples/PHP/reports/get.php
<?php
//For more information please refer to https://github.com/checkout/checkout-sdk-php
use Checkout\\CheckoutApiException;
use Checkout\\CheckoutAuthorizationException;
use Checkout\\CheckoutSdk;
use Checkout\\Environment;
use Checkout\\OAuthScope;
use Checkout\\Reports\\ReportsQuery;
use DateInterval;
use DateTime;
use DateTimeZone;

//API Keys
$api = CheckoutSdk::builder()->staticKeys()
    ->environment(Environment::sandbox())
    ->secretKey("secret_key")
    ->build();

//OAuth
$api = CheckoutSdk::builder()->oAuth()
    ->clientCredentials("client_id", "client_secret")
    ->scopes([OAuthScope::$Reports, OAuthScope::$ReportsView])
    ->environment(Environment::sandbox())
    ->build();

$created_after = new DateTime();
$created_after->setTimezone(new DateTimeZone("europe/madrid"));
$created_after->sub(new DateInterval("P7D"));

$query = new ReportsQuery();
$query->created_after = $created_after;
$query->created_before = new DateTime();

try {
    $response = $api->getReportsClient()->getAllReports($query);
} catch (CheckoutApiException $e) {
    // API error
    $error_details = $e->error_details;
    $http_status_code = isset($e->http_metadata) ? $e->http_metadata->getStatusCode() : null;
} catch (CheckoutAuthorizationException $e) {
    // Bad Invalid authorization
}

File: nas_spec/code_samples/Python/reports/get.py
# For more information please refer to https://github.com/checkout/checkout-sdk-python
import checkout_sdk
from checkout_sdk.checkout_sdk import CheckoutSdk
from checkout_sdk.environment import Environment
from checkout_sdk.exception import CheckoutApiException, CheckoutArgumentException, CheckoutAuthorizationException
from checkout_sdk.oauth_scopes import OAuthScopes
from datetime import datetime, timezone
from dateutil.relativedelta import relativedelta
from checkout_sdk.reports.reports import ReportsQuery

# API Keys
api = CheckoutSdk.builder() \\
    .secret_key('secret_key') \\
    .environment(Environment.sandbox()) \\
    .build()
    # or Environment.production()

# OAuth
api = CheckoutSdk.builder() \\
    .oauth() \\
    .client_credentials('client_id', 'client_secret') \\
    .environment(Environment.sandbox()) \\
    .scopes([OAuthScopes.REPORTS, OAuthScopes.REPORTS_VIEW]) \\
    .build()

__query = ReportsQuery()
__query.created_after = datetime.now(timezone.utc) - relativedelta(days=7)
__query.created_before = datetime.now(timezone.utc)

try:
    response = api.reports.get_all_reports(__query)
except CheckoutApiException as err:
    # API error
    request_id = err.request_id
    status_code = err.http_status_code
    error_details = err.error_details
    http_response = err.http_response
except CheckoutArgumentException as err:
    # Bad arguments

except CheckoutAuthorizationException as err:
    # Invalid authorization